packages/react-charting/src/components/TreeChart/TreeChart.base.tsx

```diff
@@ -0,0 +1,293 @@
+      // check for leaf nodes
+      if (!d.children && !parentSet.has(d.parentID) && treeHeight === 3) {
+        const newWidth = 70,
+          newHeight = 60;
+        parentSet.add(d.parentID);
+
+        let children: any = BFS[d.parentID]?.children;
+        // if the parent has 1 child
+        if (children.length === 1) {
+          this.addNodeShapetoSVG(d.dataName, d.subName, d.x, d.y, d.fill, rectWidth, newHeight);
+        }
+
+        // if the parent has more than 2 child
+
+        let dx1: number = BFS[d.parentID]?.x - newWidth * 1.6,
+          dy: number = children[0]?.y,
+          dx2: number = BFS[d.parentID]?.x;
+
+        if (children.length >= 2) {
+          for (let itr = 0; itr < children.length; ++itr) {
+            const child = children[itr];
+            // For compact compostion
+            if (this.composition === 0) {
+              if (itr % 2 === 0) {
+                this.addNodeShapetoSVG(
+                  child.data.name,
+                  child.data.subname,
+                  dx1 + gap * 4,
+                  dy,
+                  child.data.fill,
+                  newWidth,
+                  newHeight,
+                );
+              }
+              if (itr % 2 === 1) {
+                this.addNodeShapetoSVG(
+                  child.data.name,
+                  child.data.subname,
+                  dx2 + gap * 4,
+                  dy,
+                  child.data.fill,
+                  newWidth,
+                  newHeight,
+                );
+                dy += newHeight + gap / 2;
+              }
+            }
+            // For long compostion
+            else {
+              this.addNodeShapetoSVG(
+                child.data.name,
+                child.data.subname,
+                dx1 + gap * 5.5,
+                dy,
+                child.data.fill,
+                rectWidth,
+                newHeight,
+              );
+              dy += newHeight + gap / 2;
+            }
+          }
+        }
+      }
+
+      if (d.children || treeHeight === 2) {
+        this.addNodeShapetoSVG(d.dataName, d.subName, d.x, d.y, d.fill, rectWidth, rectHeight);
+      }
+    }
+
+    // <------------------ Links section ------------------>
+
+    // Create path element
+    let link = this.svg.selectAll('path.link').data(nodes.slice(1), function (d: any) {
+      return d.id;
+    });
+
+    // UPDATE the link
+    let linkUpdate = link
+      .enter()
+      .insert('path', 'g')
+      .attr('class', 'link')
+      .style('fill', 'none')
+      .style('stroke', 'black')
+      .style('stroke-width', '2px');
+
+    let linkParentSet = new Set();
+    linkUpdate.attr('d', (d: any) => {
+      if (treeHeight === 3) {
+        // leaf nodes with more than 2 sibling nodes
+        if (!d.children && !linkParentSet.has(d?.parent.id)) {
+          linkParentSet.add(d?.parent.id);
+          return this.addLinktoNodes(BFS[d.id], BFS[d.parent.id], true, rectWidth, gap);
```

Comment:
shouldn't the link be defined from parent to child?

Reply:
fixed (exchanged function arguments)